File: 937-OnlineStockSpan/937-OnlineStockSpan.java
// Last updated: 04/04/2026, 23:32:44
class StockSpanner {
    class State {
        int price;
        int span;
        State(int price,int span) {
            this.price=price;
            this.span=span;
        }
    }
    Stack<State> stack;
    public StockSpanner() {
        this.stack=new Stack<>();
    }
    public int next(int price) {
        int result=1;
        while(!stack.isEmpty() && stack.peek().price <= price) {
            result += stack.pop().span;
        }
        stack.push(new State(price,result));
        return result;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */

File: 937-OnlineStockSpan/937-OnlineStockSpan.cpp
// Last updated: 04/04/2026, 23:32:40
class StockSpanner {
    stack<vector<int>> st;
    int currDay;
    public:
    StockSpanner() {
        currDay=-1;
    } 
    int next(int price) {
        int ans=1;
        currDay++;
        while(!st.empty() && price >= st.top()[0]) {
            st.pop();
        } 
        if(st.empty()) {
            ans=currDay+1;
        } else {
            ans=currDay-st.top()[1];
        }
        st.push( {price,currDay});
        return ans;
    }
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */